File: 71-simplify-path/simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        str_1=""
        list_2=[]
        list_1=path.split('/')
        for i in list_1:
            if i=='' or i=='.':
                continue
            if i=='..':
                if list_2:
                    list_2.pop()
            else:
                list_2.append(i)
        print("/".join(list_2))
        return "/"+"/".join(list_2)

File: 71-simplify-path/simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] sa= path.split("/");
        Stack<String> st=new Stack<>();

        for(String c:sa)
        {
            if(c.equals("") || c.equals("."))
            {
                continue;
            }
            if(c.equals(".."))           
            {
               
                if(!st.isEmpty() )
            {
                st.pop();
            }
            }
            else
            {
                st.push(c);
            }
        }
        StringBuilder sb=new StringBuilder();

        while(!st.isEmpty())
        {
            sb.insert( 0,"/" + st.pop());
        }
        if(sb.length()==0)
            return "/";
         return sb.toString();
    }
}